### PR TITLE
job_scripts - POST

### PR DESCRIPTION
**Pull request by zsinx6**
Working with the https://github.com/omnivector-solutions/jobbergate-cli/tree/local_fastapi so it is possible to just export the correct URL then test from there.

The current cyclomatic complexity is 19, the goal of the next PR is to lower to at least 15.

### CODE REVIEW

File: jobbergateapi2/apps/job_scripts/routers.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    except KeyError:
+        supporting_files = []
+
+    default_template = [
+        default_template := param_dict["jobbergate_config"].get("default_template"),
+        "templates/" + default_template,
+    ]
+
+    application_location = (
+        f"{settings.S3_BASE_PATH}/TEST/applications/{application.id}/jobbergate.tar.gz"
```
Comment (zsinx6) [inline jobbergateapi2/apps/job_scripts/routers.py:91]: This is here for testing only, the line below will be used

File: jobbergateapi2/apps/job_scripts/routers.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+from jobbergateapi2.config import settings
+from jobbergateapi2.storage import database
+
+S3_BUCKET = f"jobbergate-api-{settings.SERVERLESS_STAGE}-{settings.SERVERLESS_REGION}-resources"
+router = APIRouter()
+
+
+def inject_sbatch_params(job_script_data_as_string: str, sbatch_params: List[str]) -> str:
+    """
+    Given the job script as job_script_data_as_string, inject the sbatch params in the correct location
```
Comment (jamesbeedy) [inline jobbergateapi2/apps/job_scripts/routers.py:31]: Possible rewording

```python
Inject sbatch params into job script.

Given the job script as job_script_data_as_string, inject the sbatch params in the correct location.
```
```

File: jobbergateapi2/apps/job_scripts/routers.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        if member.name in default_template:
+            contentfobj = s3_application_tar.extractfile(member)
+            template_files["application.sh"] = contentfobj.read().decode("utf-8")
+        if member.name in supporting_files:
+            match = [x for x in support_files_ouput if member.name in x]
+            contentfobj = s3_application_tar.extractfile(member)
+            filename = support_files_ouput[match[0]][0]
+            template_files[filename] = contentfobj.read().decode("utf-8")
+
+    # Use tempfile to generate .tar in memory - NOT write to disk
```
Comment (jamesbeedy) [inline jobbergateapi2/apps/job_scripts/routers.py:112]: 💯

File: jobbergateapi2/apps/job_scripts/schemas.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+"""
+Defines the schema for the JobScript resource
```
Comment (jamesbeedy) [inline jobbergateapi2/apps/job_scripts/schemas.py:2]: JobScript resource schema.

File: jobbergateapi2/tests/apps/applications/__init__.py [hunk 5]
```diff
@@ -0,0 +1,3 @@
+"""
+Tests for the Application resource
```
[the remark this answers is not in the document]
Comment (zsinx6) [reply jobbergateapi2/tests/apps/applications/__init__.py:2]: Also fixed in the `tests/apps/job_scripts/__init__.py`

File: jobbergateapi2/apps/job_scripts/models.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+"""
+Database model for the JobScript resource
```
Comment (jamesbeedy) [inline jobbergateapi2/apps/job_scripts/models.py:2]:
```suggestion
Database model for the JobScript resource.
```

File: jobbergateapi2/apps/job_scripts/routers.py [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    job_script_name: str = Form(...),
+    job_script_description: Optional[str] = Form(""),
+    application_id: int = Form(...),
+    current_user: User = Depends(get_current_user),
+    upload_file: UploadFile = File(...),
+    sbatch_params: Optional[List[str]] = Form(None),
+    param_dict: Optional[str] = Form(None),
+):
+    """
+    Create new job_script using an authenticated user.
```
Comment (jamesbeedy) [inline jobbergateapi2/apps/job_scripts/routers.py:60]:
```suggestion
    Create a new job script.
    
    Make a post request to this endpoint with the required values to create a new job script.
```

File: jobbergateapi2/apps/job_scripts/schemas.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+"""
+Defines the schema for the JobScript resource schema.
```
Comment (jamesbeedy) [inline jobbergateapi2/apps/job_scripts/schemas.py:2]:
```suggestion
JobScript resource schema.
```

File: jobbergateapi2/tests/apps/conftest.py [hunk 9]
```diff
@@ -31,3 +31,15 @@ def user_data():
         "username": "username",
         "password": "supersecret123456",
     }
+
+
+@fixture
+def application_data():
+    """
+    Default appliation data for testing
```
Comment (jamesbeedy) [inline jobbergateapi2/tests/apps/conftest.py:39]:
```suggestion
    Define the default application data for testing.
```

File: jobbergateapi2/tests/apps/job_scripts/conftest.py [hunk 10]
```diff
@@ -0,0 +1,30 @@
+from pytest import fixture
+
+
+@fixture
+def job_script_data():
+    """
+    Default job_script data for testing
```
Comment (jamesbeedy) [inline jobbergateapi2/tests/apps/job_scripts/conftest.py:7]:
```suggestion
    Default job_script data for testing.
```

**Review by jamesbeedy — APPROVED** [review]
👍

**Pull request merged**